### PR TITLE
[CARBONDATA-2529] Fixed S3 Issue for Hadoop 2.8.3

### PR DESCRIPTION
**Pull request by bhavya411**
This issue fixes the issue while loading the data with S3 as backend.

 - [ N] Any interfaces changed?
 
 - [N ] Any backward compatibility impacted?
 
 - [ N] Document update required?

 - [Y ] Testing done

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6099/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4937/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5093/

**Comment by chenliang613** [comment]
LGTM.  is there any other URI for object storage?

File: core/src/main/java/org/apache/carbondata/core/datastore/filesystem/HDFSCarbonFile.java [hunk 1]
```diff
@@ -120,7 +120,8 @@ public boolean renameForce(String changetoName) {
         ((DistributedFileSystem) fs).rename(fileStatus.getPath(), new Path(changetoName),
             org.apache.hadoop.fs.Options.Rename.OVERWRITE);
         return true;
-      } else if (fileStatus.getPath().toString().startsWith("s3n")) {
+      } else if (fileStatus.getPath().toString().startsWith("s3n") || fileStatus.getPath()
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/filesystem/HDFSCarbonFile.java]: move `fileStatus.getPath()` to next line

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5043/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6206/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5179/

**Pull request closed without merging**